### PR TITLE
#16 Added unit tests

### PR DESCRIPTION
**Pull request by LarryBattle**
Bind the VueModel instance to `Vue.models.register()`. Otherwise, the user will get one of these errors when calling `Vue.models.register()`.

```
Uncaught TypeError: Cannot read property 'register' of undefined
```
or
```
TypeError: Cannot set property 'users' of undefined
```

If you run the tests in the PR without the changes to `VueModel.js`, you'll get this output.

Terminal
```$> npm test```

Output:
``` VueModel
    Vue.models.register() should
      ✖ register model(s)
      ✔ register model(s) when Vue.use(VueModel, options) is called

Finished in 0.003 secs / 0.038 secs @ 02:35:35 GMT-0500 (CDT)

SUMMARY:
✔ 1 test completed
✖ 1 test failed

FAILED TESTS:
  VueModel
    Vue.models.register() should
      ✖ register model(s)
        Chrome 61.0.3163 (Linux 0.0.0)
      TypeError: Cannot set property 'users' of undefined
          at Object.register (/tmp/ed43830fc058edb9203b0602f1a56f77.browserify:28533:29)
          at UserContext.it (/tmp/ed43830fc058edb9203b0602f1a56f77.browserify:28723:24)
```

## Changes
- Added unit tests for `VueModel.js` in `VueModel.spec.js`.
- Added [karma test runner](
https://karma-runner.github.io/1.0/index.html)
- Added [jasmine for unit testing](https://jasmine.github.io/)
- Added package-lock.json for [added security](https://docs.npmjs.com/files/package-lock.json)
- Updated docs with testing notes.

## Related PRs
- https://github.com/aarondfrancis/vue-model/issues/16
- https://github.com/aarondfrancis/vue-model/issues/18

### CODE REVIEW

File: src/VueModel.js [hunk 1]
```diff
@@ -188,11 +188,11 @@ class VueModel {
         });
 
         Vue.models = {
-            register: this.register,
+            register: this.register.bind(this),
```
Comment (shealan) [inline src/VueModel.js:191]: Is this what was causing the problems using this package?

Comment (LarryBattle) [reply src/VueModel.js:191]: Yes. I was able to create a simple demo after adding this fix.

**Comment by sne11ius** [comment]
Merge plz :D

**Comment by aarondfrancis** [comment]
Merged! Thanks a million @LarryBattle.

**Pull request merged**